src/breeze/components/breeze/sidebar_links.cr

```diff
@@ -1,19 +1,32 @@
 class Breeze::SidebarLinks < BreezeComponent
   needs context : HTTP::Server::Context
 
-  # NOTE: https://github.com/luckyframework/lucky/issues/1243
-  # Habitat.create do
-  #   setting links : Hash(String, BreezeAction.class) = {
-  #     "Requests" => Breeze::Requests::Index,
-  #     "Queries"  => Breeze::Queries::Index,
-  #   }
-  # end
+  Habitat.create do
+    setting links : Array(Proc(Breeze::SidebarLinks, Nil)) = [
+      ->(breeze : Breeze::SidebarLinks) {
+        breeze.mount(Breeze::SidebarLink, context: breeze.context, link_text: "Requests", link_to: Breeze::Requests::Index.path)
+      },
+      ->(breeze : Breeze::SidebarLinks) {
+        breeze.mount(Breeze::SidebarLink, context: breeze.context, link_text: "Queries", link_to: Breeze::Queries::Index.path)
+      },
+    ]
+  end
```

Comment:
Yeah, we'd have to make a new `mount` method that takes a component instance. I think that would be cool though. Do you think this is fine as is until I can release a new Lucky? I could always come back and update later.

Reply:
The more I dig in to this, the more I fight with the implementation. The above suggested change wouldn't work for a few reasons. 1. It needs a return type set. If it's `Nil`, then it wouldn't be allowed to be passed in to a `mount`, or even have methods called off it (i.e. `component.call(self).view(view).render`). 2. With how strict the Crystal `Proc` is, you can't define a base type, and allow it to return any sub-type. This doesn't compile:

```crystal
setting links : Array(Proc(Breeze::NavbarLinks, Lucky::BaseCompoent)) = [
  ->(breeze : Breeze::NavbarLinks) {
     Breeze::NavbarLink.new(...)
  }
]
```

You end up getting the error:

```
web          |  5 | setting links : Array(Proc(Breeze::NavbarLinks, Lucky::BaseComponent)) = [
web          |                                                                               ^
web          | Error: class variable '@@links' of Breeze::NavbarLinks::HabitatSettings must be (Array(Proc(Breeze::NavbarLinks, Lucky::BaseComponent)) | Nil), not (Array(Proc(Breeze::NavbarLinks, Breeze::NavbarLink)) | Nil)
web          |
```

Now, if we return just a component instance, and it *has* to be a `Breeze::NavbarLink` instance, then the only reason we're passing in the `breeze` instance is to get access to `context`. We could shorten this whole structure by just asking "What's the title of your link, and where does it go?". The issue there is that it would be defined as `Tuple(String, String)` which doesn't really give you much information. Maybe it could be aliased as `Tuple(LinkText, LinkLocation)`, but I'm not sure how well Crystal will play with the aliases there 🤔 

The alternative is if we keep the definition as is, you have control to mount any component you'd like. The downside is that it's a little weird you have to call `breeze.mount`, and you have to know that everything is scoped to breeze `breeze.context`.

```crystal
# setting links : Array(Proc(Breeze::NavbarLinks, Nil))
# This works with the code as is now.
Breeze::NavbarLinks.settings.links << ->(breeze : Breeze::NavbarLinks) {
  breeze.mount MyComponent
}
Breeze::NavbarLinks.settings.links << ->(breeze : Breeze::NavbarLinks) {
  breeze.mount Breeze::NavbarLink, ....
}
``` 

```crystal
class MyComponent < Lucky::BaseComponent
  def render
    div "Static"
  end
end
```

<img width="335" alt="Screen Shot 2021-03-12 at 9 18 45 AM" src="https://user-images.githubusercontent.com/2391/110974935-f8664000-8313-11eb-8de1-f6d8f7220caf.png">

Reply:
The beauty of using Crystal classes, is we can just make a method that does this for us! Not sure why I didn't think of this before, but now we have this:

```crystal
# Use what we have built-in
Breeze::NavbarLinks.settings.links << ->(navbar : Breeze::NavbarLinks) {
  navbar.mount_link("Emails", to: Breeze::Emails::Index)
}

# Escape hatch to use your own custom component!
# This also means you can have a dropdown menu 😁 
Breeze::NavbarLinks.settings.links << ->(navbar : Breeze::NavbarLinks) {
  navbar.mount(MyCustomComponent, context: navbar.context, styles: "whatever")
}
```